Sources/NIO/EventLoop.swift

```diff
@@ -273,13 +273,24 @@ public protocol EventLoop: EventLoopGroup {
     /// Contrary to `makeSucceededFuture`, `makeSucceededVoidFuture` is a customization point for `EventLoop`s which
     /// allows `EventLoop`s to cache a pre-succeded `Void` future to prevent superfluous allocations.
     func makeSucceededVoidFuture() -> EventLoopFuture<Void>
+
+    /// Must crash if it is not safe to call `wait()` on an `EventLoopFuture`.
+    ///
+    /// This method is a debugging hook that can be used to override the behaviour of `EventLoopFuture.wait()` when called.
+    /// By default this simply becomes `preconditionNotInEventLoop`, but some `EventLoop`s are capable of more exhaustive
+    /// checking and can validate that the wait is not occuring on an entire `EventLoopGroup`, or even more broadly.
+    func _preconditionSafeToWait(file: StaticString, line: UInt)
```

Comment:
Why do we want to underscore this method?

Reply:
Because we don't require implementing it and we don't want anyone to call it directly.

Reply:
Can we document that we don't want anyone to call it directly?

Reply:
Done.